src/sentry/incidents/logic.py

```diff
@@ -936,22 +936,37 @@ def update_alert_rule(
                 raise ResourceDoesNotExist(
                     "Your organization does not have access to this feature."
                 )
-
             if updated_fields.get("detection_type") == AlertRuleDetectionType.DYNAMIC and (
                 alert_rule.detection_type != AlertRuleDetectionType.DYNAMIC or query or aggregate
             ):
                 for k, v in updated_fields.items():
                     setattr(alert_rule, k, v)
 
+                for k, v in updated_query_fields.items():
+                    if k == "dataset":
+                        v = v.value
+                    elif k == "time_window":
+                        v = time_window if time_window else snuba_query.time_window
+                    elif k == "event_types":
+                        continue
+                    setattr(alert_rule.snuba_query, k, v)
+
                 try:
                     # NOTE: if adding a new metric alert type, take care to check that it's handled here
                     rule_status = send_historical_data_to_seer(
                         alert_rule=alert_rule,
                         project=projects[0] if projects else alert_rule.projects.get(),
+                        snuba_query=alert_rule.snuba_query,
```

Comment:
since we fetch the snuba query from the db in `send_historical_data_to_seer` it was missing the updates unless I passed it like this